### PR TITLE
Adopt exists unit tests for using in TypeScript

### PR DESCRIPTION
**Pull request by akazakou**
### Changelog
* Copy and adopt exists unit tests for a run under TypeScript control

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -679,7 +680,7 @@ declare module 'fastest-validator' {
 	/**
 	 * Description of validation rule definition for a some property
 	 */
-	type ValidationRule = ValidationRuleObject | ValidationRuleObject[] | ValidationRuleName | string;
+	type ValidationRule = ValidationRuleObject | ValidationRuleObject[] | ValidationRuleName | number;
```
Comment (icebob) [inline index.d.ts]: Why `number`?

Comment (akazakou) [reply index.d.ts]: Nice catch. I've tested negative cases for `v.compile(1)` to throw errors

Comment (icebob) [reply index.d.ts]: Travis failed :(

Comment (akazakou) [reply index.d.ts]: Yeah. Some last time moment fixes :(

Comment (akazakou) [reply index.d.ts]: Now it's green

**Pull request merged**